libr/include/r_types.h

```diff
@@ -570,11 +570,15 @@ enum {
 	R_SYS_ARCH_LM32 =  0x80000000LL, // 1<<31
 };
 
+#if !defined(__HAIKU__)
```

Comment:
move this condition in the line below like: `#if !defined(__HAIKU__) && HAVE_CLOCK_NANOSLEEP ...

Reply:
I did because of CLOCK_MONOTONIC value on Haiku

Reply:
What do u mean?

Reply:
CLOCK_MONOTONIC is 0

Reply:
i dont get it. if clock-monotonic is 0, then clock-nanosleep will be 0 for haiku, which is what you are trying to do in this ifdef, right?. so maybe the `if !defined(__HAIKU__)` is not necessary?